etcdserver/server.go

```diff
@@ -669,6 +669,10 @@ func (s *EtcdServer) apply(es []raftpb.Entry, confState *raftpb.ConfState) (uint
 		e := es[i]
 		switch e.Type {
 		case raftpb.EntryNormal:
+			// raft module may generate empty Entry
```

Comment:
// raft state machine may generate noop entry when leader switches. skip it in advance to avoid some potential bug in the future

It is good to have verbose comments here because it is not straightforward.